crypto/bio/bss_dgram.c

```diff
@@ -195,68 +194,54 @@ static void dgram_adjust_rcv_timeout(BIO *b)
 {
 # if defined(SO_RCVTIMEO)
     bio_dgram_data *data = (bio_dgram_data *)b->ptr;
+    OSSL_TIME timeleft;
 
     /* Is a timer active? */
-    if (data->next_timeout.tv_sec > 0 || data->next_timeout.tv_usec > 0) {
-        struct timeval timenow, timeleft;
-
+    if (!ossl_time_is_zero(data->next_timeout)) {
         /* Read current socket timeout */
 #  ifdef OPENSSL_SYS_WINDOWS
         int timeout;
```

Comment:
The type incompatibility on Windows on line 236 is probably best resolved by changing this to a `uint64_t`, and then have a temporary variable (say, `int getsockopt_arg`) specifically for `getsockopt()`, and follow up that call with an assignment to `timeout` (`timeout = getsockopt_arg`) following that call.  Or something like that...

It sucks to have to do that sort of thing, but...

Reply:
A cast is usually sufficient to remove the warning.

Reply:
Sure, I'm just worried about the loss of data.  But, if you know that the `..2ms()`  call will never return 2^(8*sizeof(int)) or higher, then it's fine

Reply:
49 days is a long timeout.

Even if it is problematic, this isn't introducing a regression -- the same thing will happen with the existing code.

Reply:
Fair enough